### PR TITLE
NMC-127 - File action button behaviour for different viewport

### PR DESCRIPTION
**Pull request by TSI-kavitasonawane**
**File action button behaviour changes:**
We now show more single actions depending on the viewport size in the table header beginning from the first option in the dropdown.
We add a new option called "Cancel" (ger: Abbrechen") which will remove any selection (just remove selection, not the files of course).
We right align the Actions button.
We will show every option in the action button dropdown. It doesnt matter which option is also shown in the table header.
We add a new label called "All".
on S: we hide the label "All" according to the screen design

Screenshot after changes:
file action

### CODE REVIEW

File: apps/files/js/app.js [hunk 1]
```diff
@@ -119,6 +119,12 @@
 							iconClass: 'icon-tag',
 							order: 100,
 						},
+						{
+							name: 'cancel',
+							displayName:  'Cancel',
+							iconClass: 'icon-cancel',
```
Comment (juliushaertl) [inline apps/files/js/app.js:125]:
```suggestion
							iconClass: 'icon-close',
```

File: apps/files/css/files.scss [hunk 2]
```diff
@@ -1238,3 +1237,91 @@ table.dragshadow td.size {
...
+
+table.multiselect {
+	thead {
+		#selectedMenu {
+			width: 100%;
+		}
+	}
+}
+
+.icon-cancel {
```
Comment (juliushaertl) [inline apps/files/css/files.scss:1261]: No need for an extra icon as we already have icon-close for this: https://github.com/nextcloud/server/pull/28834/files#r721064824

**Comment by juliushaertl** [comment]
Some screenshots for before/after would be nice to make this easier to review for our design team :)

**Review by artonge — CHANGES_REQUESTED** [review]
I like this change a lot, but I have some design feedback:

## There is a flash when selecting or deselecting an element

Peek 2021-10-04 12-54

## Can we have some additional style on `hover`?

Some ugly ideas:

Screenshot from 2021-10-04 12-52-06
OR
Screenshot from 2021-10-04 12-52-23

## Can we align "All" and "1 file (164 B)"
Screenshot from 2021-10-04 12-46-17

## Can we make the "Cancel" button different to highlight that it is not an action like the others?
Something like that with the cross also styled in red:

Screenshot from 2021-10-04 12-58-50

File: apps/files/css/files.scss (hunk 2 above)
Comment (TSI-kavitasonawane) [reply apps/files/css/files.scss]: @juliushaertl  I have done the changes

**Comment by TSI-kavitasonawane** [comment]
> I like this change a lot, but I have some design feedback:
> 
> ## There is a flash when selecting or deselecting an element
> Peek 2021-10-04 12-54
> 
> ## Can we have some additional style on `hover`?
> Some ugly ideas:
> 
> Screenshot from 2021-10-04 12-52-06 OR Screenshot from 2021-10-04 12-52-23
> 
> ## Can we align "All" and "1 file (164 B)"
> Screenshot from 2021-10-04 12-46-17
> 
> ## Can we make the "Cancel" button different to highlight that it is not an action like the others?
> Something like that with the cross also styled in red:
> 
> Screenshot from 2021-10-04 12-58-50

@artonge  I have done all the changes which you suggested. Please check it.

**Comment by artonge** [comment]
> @artonge I have done all the changes which you suggested. Please check it.

Thanks, this looks much better already :)

Three new nitpicks:

1. The trash icon is red on hover, can we prevent that to make it more persistent with other actions ? Or maybe go further and make it red when the label is hovered too. Maybe make the label red too.

Screenshot from 2021-10-05 11-04-05

2. Should we move the Cancel action outside of the three dots menu and have it always accessible ?

3. Should we only list the hidden actions in the three dots menu instead of listing them all ?

Screenshot from 2021-10-05 11-10-39

Maybe we should @jancborchardt for some proper design inputs :)

**Comment by TSI-kavitasonawane** [comment]
> > @artonge I have done all the changes which you suggested. Please check it.
> 
> Thanks, this looks much better already :)
> 
> Three new nitpicks:
> 
> 1. The trash icon is red on hover, can we prevent that to make it more persistent with other actions ? Or maybe go further and make it red when the label is hovered too. Maybe make the label red too.
> 
@artonge I have done the changes which you suggested for trash icon.
> Screenshot from 2021-10-05 11-04-05
> 
> 1. Should we move the Cancel action outside of the three dots menu and have it always accessible ?

We can't do this because we don't have space for Cancel action in small screen.
Table Header Functions S@2x
 
> 2. Should we only list the hidden actions in the three dots menu instead of listing them all ?
>  As per requirement we have to show every option in the action button dropdown. It doesnt matter which option is also shown in the table header.  @jancborchardt  Please confirm this.
> Screenshot from 2021-10-05 11-10-39
> 
> Maybe we should @jancborchardt for some proper design inputs :)

**Comment by jancborchardt** [comment]
Agree with all of @nimishavijay’s feedback above – @TSI-kavitasonawane anything remaining from those items? :)

**Comment by blizzz** [comment]
changing the milestone to 24 for the PR is against master, a feature and we branch off 23 very soon.

**Comment by juliushaertl** [comment]
@skjnldsv All covered in https://github.com/nextcloud/server/pull/37642 right? Just to keep a reference here as well

**Pull request closed without merging**